packages/datatrak-web/src/features/Survey/SurveyLayout.tsx

```diff
@@ -87,17 +88,21 @@ export const SurveyLayout = () => {
     visibleScreens,
     isResubmitReviewScreen,
   } = useSurveyForm();
+
   const { handleSubmit, getValues } = useFormContext();
   const { mutate: submitSurveyResponse, isLoading: isSubmittingSurveyResponse } =
-    useSubmitSurveyResponse(from);
+    useSubmitSurveyResponse({ from, primaryEntityCode });
   const { mutate: resubmitSurveyResponse, isLoading: isResubmittingSurveyResponse } =
     useResubmitSurveyResponse();
   const { back, next } = useSurveyRouting(numberOfScreens);
 
   const handleStep = (path, data) => {
     updateFormData({ ...formData, ...data });
     navigate(path, {
-      state: { ...(from && { from }) },
+      state: {
+        ...(from && { from }),
+        ...(primaryEntityCode && { primaryEntityCode }),
```

Comment:
I wonder if we can make some sort of hook or something that handles the location state since we do this in a couple of places